File: r/prime-factors/prime-factors.R
prime_factors <- function(n) {
  factors <- c()
  f <- 2
  while (f ^ 2 <= n) {
    if (n %% f == 0) {
      factors <- append(factors, f)
      n <- n / f
    } else {
      f <- f + 1
    }
  }
  if (n > 1) factors <- append(factors, n)
  factors
}

File: dart/prime-factors/lib/prime_factors.dart
class PrimeFactors {
  List<int> factors(int number) {
    var fs = <int>[];
    var factor = 2;

    while (factor * factor <= number) {
      if (number % factor == 0) {
        fs.add(factor);
        number ~/= factor;
      } else {
        factor++;
      }
    }

    if (number > 1) fs.add(number);
    return fs;
  }
}
